### ISSUE TITLE
Openbay Public Database [2]

### ISSUE THREAD

**Issue by TPBT-OFFICIAL** [isohuntto/openbay#142]
MYSQL IP: 5.135.13.72
MYSQL PORT: 3306
MYSQL Username: openbay
MYSQL Password: AcSxMV4AEpKFEWK5

SPHINX IP: 5.135.13.72
SPHINX PORT: 9306

Im not sure if this is how it works but since i got mine up and running i might as well give you all access to it and use it on your website :) i don't know how fast this is but if anyone can help me optimise it or make it faster i would apreciate it!
I would also apreciate any donation: https://tpbt.org/donate.php

**Comment by Poisonleaf** [comment]
Imma connect my page to that public DB and see how it goes.
keep ya posted!

**Comment by wacgyver** [comment]
Which file are you editing with these adresses?

**Comment by TPBT-OFFICIAL** [comment]
@wacgyver you should edit the config file in "protected/config/config.php"

**Comment by wacgyver** [comment]
Thank you! yes i figured out by searching for the these parameters, but it doesent work after changing to what you suggest either. I must have a different problem, because im getting this error message whatever i do:  (only the front page seems to look fine, but nothing actually works. Allthough i can of course make a "browse" folder and put torrents into it, and it'll work, but i dont think that is the way its supposed to work)

Not Found
The requested URL /src/www/search was not found on this server.
Apache/2.4.10 (Ubuntu) Server at 10.0.0.101 Port 80

**Comment by wacgyver** [comment]
Im using apace and a standard installation, no customization done on either box (box1 Debian, Box2 Ubuntu). Std installation of php also. Do i need to customize this to work? PHP obviously works, but to i have to do something to PHP-FPM? Im not an expert, so id greatly appreciate input as to what is actually needed and what may be the cause of my problem. 
Should i use Nginx instead, and in case why so?

**Comment by TPBT-OFFICIAL** [comment]
you should just use the folder "www" as the main folder of the website and you should have "protected"  one directory back like this:
/[somelocation]/www/
/[somelocation]/protected/ (only these two folders)
Then it should work properly i think.

**Comment by wacgyver** [comment]
I have also tried that, right now and several times before. This is from my Ubuntu box right now: 
http://10.0.0.101/search?q=test
Not Found
The requested URL /search was not found on this server.
Apache/2.4.10 (Ubuntu) Server at 10.0.0.101 Port 80

"www" folder used as the main folder of the website "protected" one level up like you said. How can i verify that PHP-fpm actually works? Do i need to change socket or something? Im not sure what php-fpm does really, that php doesent do. Is it just more efficient for something? Is is not really needed? Im confused as to what could be wrong. I dont get the 404 that others get when the hosted DB server is offline, it seems to be something that fails locally. Is there any way any of could make an image of a working hosted installation either on VMware or Virtualbox? That would problaby be a lot smaller than the a hosted installation with db, and besides id really like to figure out what my problem is here.. This should be the easiest setup, and yet i cant get anything openbay related working here .. while everything else sees to work great

**Comment by TPBT-OFFICIAL** [comment]
I think i know why.... do you have it configured to allow .htaccess files?

**Comment by wacgyver** [comment]
damn... i dont know. Probably not then, unless its allowed by default. Ill try to find the config for that now. Is is in apache.conf?

**Comment by TPBT-OFFICIAL** [comment]
well if its ubuntu then yeah but i don't know a lot about ubuntu so i don't know what to look for but im sure its because of that.

**Comment by wacgyver** [comment]
These are the files with "htaccess" in them i think:
root@tpbuntu:/etc# grep -rnw 'apache2' -e "htaccess"
apache2/mods-available/php5.conf:22:# prevents .htaccess files from disabling it.
apache2/apache2.conf:183:AccessFileName .htaccess
apache2/apache2.conf:186:# The following lines prevent .htaccess and .htpasswd files from being

**Comment by wacgyver** [comment]
I have excactly the same problem in Debian though.

**Comment by mywish1** [comment]
i have upload these files in my domain i mean i upload 2 folders only 1 of www and 1 is protected folder i move it from src folder to my main domain i mean like /[somelocation]/www/
/[somelocation]/protected/ but still my site is not working even give me 500 internal server error what i did wrong ? plz help me thanks i setup sphinx as well

**Comment by wacgyver** [comment]
This is pretty much it .. htaccess is mentioned 2 times in apache2.conf, 1 for which file to look for and 1 for "prevent .htaccess and .htpasswd files from being viewed by Web clients" 
This is all the default setup. I have a hard time understanding that i should need to change this setup to get openbay working... do you still think this is the problem? (i mean .htaccess settings )?


# AccessFileName: The name of the file to look for in each directory
# for additional configuration directives.  See also the AllowOverride
# directive.
#
AccessFileName .htaccess

#
# The following lines prevent .htaccess and .htpasswd files from being
# viewed by Web clients.
#
<FilesMatch "^\.ht">
        Require all denied
</FilesMatch>

**Comment by wacgyver** [comment]
Shit... sorry about the strange formatting

**Comment by wacgyver** [comment]
I have not set up dns. Should that be even remotely related to the 

root@tpbuntu:/etc# /etc/init.d/apache2 restart
 * Restarting web server apache2                                                                       AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 127.0.1.1. Set the 'ServerName' directive globally to suppress this error message i get?
(Not Found
The requested URL /search was not found on this server.)

**Comment by TPBT-OFFICIAL** [comment]
no you just need to change all "AllowOverride None" to "AllowOverride All" just look for a tutorial on google

**Comment by wacgyver** [comment]
Yes it works now after changing  the "AllowOverride" from None to All :+1: :)

But only on my Ubuntu box, the Debian box did not like the change and came up with "Internal Error" whenever the  "AllowOverride" was set to All

**Comment by Lacedaemon** [comment]
@TPBT-OFFICIAL: Thanks for sharing your database and Sphinx!  They're pretty slow, but I can confirm that they work.  As stated [here](https://github.com/isohuntto/openbay/issues/142#issuecomment-70795782), just edit `tpb/protected/config/config.php` and change the appropriate lines according to the OP.

Finally, my OpenBay has some life for a change!

**Comment by TPBT-OFFICIAL** [comment]
@wacgyver If in the debian box it gave a "internal error" then it means the rewrite module isn't installed i think.... :P

**Comment by TPBT-OFFICIAL** [comment]
@Lacedaemon Your welcome! Im trying to get a dedicated server for this when i do i promise ill update this post :)

**Comment by Lacedaemon** [comment]
@TPBT-OFFICIAL: Take your time.  You've already done more than Isohunt has in the recent past. <_<

**Comment by wacgyver** [comment]
@TPBT-OFFICIAL Yes youre right again. (rewrite module was not installed)
Now it works fine too!  Thanks a lot for being very helpful ! - ]

**Comment by wacgyver** [comment]
Is there a good guide here for setting up an openbay site with local spinxsearch and DB?

**Comment by Lacedaemon** [comment]
@wacgyver: See [here](https://github.com/isohuntto/openbay/issues/129).

**Comment by yatznet** [comment]
@TPBT-OFFICIAL i have set my copy of openbay up using the information in the original post and it half works. i can access you list of torrents on the recent torrents page but search and browse show up 404 if you can offer any assistance on this it would be greatly appreciated the site url is http://flag.xmweb.biz its not going to be its perminent home im using it for testing

**Comment by Lacedaemon** [comment]
@yatznet: I may be of assistance.

The absence of search means there must be something wrong with your Sphinx configuration.  Are you sure you configured Sphinx as seen above in `config.php`?  Here's how lines 30-34 of my `config.php` look like:

    'sphinx' => array(
        'class' => 'system.db.CDbConnection',
        'connectionString' => 'mysql:host=5.135.13.72;port=9306',
        'queryCacheID' => 'cache'
    ),

**Comment by yatznet** [comment]
@Lacedaemon heres mine
        'sphinx' => array(
            'class' => 'system.db.CDbConnection',
            'connectionString' => 'mysql:host=5.135.13.72;port=9306',
            'queryCacheID' => 'cache'
        ),
but this appears to be exactly the same as yours so im still none the wiser to the cause of the issue

**Comment by yatznet** [comment]
debug mode give me 
 CDbConnection failed to open the DB connection: SQLSTATE[HY000] [2003] Can't connect to MySQL server on '5.135.13.72' (111)

/home2/yatterz/public_html/flag/src/protected/extensions/ESphinxQL/ESphinxQL.php(297)

line 297 looks like

  $this->_search = Yii::app()->sphinx->quoteValue($search);

**Comment by Lacedaemon** [comment]
@yatznet: Thanks for the debug output, that's way more informative.

So it's the MySQL DB that's the issue... Here's how my `db` string looks like:

    'db' => array(
        'connectionString' => 'mysql:host=5.135.13.72;port=3306;dbname=tpbt',
        'username' => 'openbay',
        'password' => 'AcSxMV4AEpKFEWK5',
        'schemaCachingDuration' => '86400',
        'charset' => 'utf8',
        'enableProfiling' => true
    ),

Did you remember to change the value of `port=`?  It's 443 by default.

**Comment by yatznet** [comment]
once again it appears we got identical settings 
        'db' => array(
            'connectionString' => 'mysql:host=5.135.13.72;port=3306;dbname=tpbt',
            'username' => 'openbay',
            'password' => 'AcSxMV4AEpKFEWK5',
            'schemaCachingDuration' => '86400',
            'charset' => 'utf8',
            'enableProfiling' => true
        ),

im suddenly wondering if it has anything to do with me being behind cloudflare

**Comment by yatznet** [comment]
besides if it has anything to do with the database this page wouldn't work http://flag.xmweb.biz/recent

**Comment by Lacedaemon** [comment]
@yatznet:

> im suddenly wondering if it has anything to do with me being behind cloudflare

That could be it, although this point I'm out of ideas.  I'm as new to this as you are.

**Comment by yatznet** [comment]
i'll try it without cloudflare and get back to u once nameservers update

**Comment by Lacedaemon** [comment]
@yatznet:

![](http://i.imgur.com/pRre9Yl.gif)

**Comment by TPBT-OFFICIAL** [comment]
@Lacedaemon @yatznet Cloudflare doesn't affect connections with the script, are you using a shared hosting or a vps/dedicated server? BTW it has to be with sphinx as the script connects to sphinx like it would connect to mysql. Now if your in a shared hosting it might be the hosting blocking the connection since its not a normal port you would connect to so you have to try to contact them, if you are in a vps or dedicated server then try to disable iptables (TEMPORARLY) and try again :)

**Comment by yatznet** [comment]
@TPBT-OFFICIAL your right it has nothing todo with cloudflare i've disabled it and it still doesn't work. and yes im on shared hosting. so let me see if i understand you. are you saying i should contact my hosting and ask them to allow access via port 9306?

**Comment by TPBT-OFFICIAL** [comment]
@yatznet Yes you should tell them to allow incoming and outgoing conections on port 9306

**Comment by yatznet** [comment]
i've opened up a ticket with my host but im almost certain there going to tell me they can only open outbound. fingers crossed

**Comment by wacgyver** [comment]
@TPBT-OFFICIAL  Is your hosting still up ?  i mean db and search. Suddely none of my boxes works ... find not torrents i mean.
They are:
http://mypiratebay.ddns.net/
and
http://flv4.ddns.net/
both set up to use your hosted search and db.

**Comment by TPBT-OFFICIAL** [comment]
@wacgyver It seems it was some kind of bug.... seems to be working now :)

**Comment by TheChiefCoC** [comment]
@TPBT-OFFICIAL Great work you are doing there bro :) you should open a GitHub repository and make a modified version of OpenBay that works :smile:

**Comment by TPBT-OFFICIAL** [comment]
@TheChiefCoC Wish i knew how to work with github :P

**Comment by TheChiefCoC** [comment]
@TPBT-OFFICIAL I can help you with that :)

**Comment by yatznet** [comment]
@TPBT-OFFICIAL it took like 5 days but my host finally got back to me and has agreed to open the 9306 port for outgoing only. not incoming. luckily for me the site works fine with just outgoing.thank you for all your support :)

**Comment by TPBT-OFFICIAL** [comment]
@yatznet Your welcome :)

**Comment by samuel9554** [comment]
@TPBT-OFFICIAL I see  you have solved the issue with the age of the torrent in /recent can you help us solve this issue.

Thank you for your public database

**Comment by TPBT-OFFICIAL** [comment]
@samuel9554 I will try to help you in any way but whats the problem? :)

**Comment by samuel9554** [comment]
Go check those picture you will understand.  This is my website : http://screencast.com/t/Xse2BCj9vv
This is your website : http://screencast.com/t/uXcYolhjyZ
I will very appreciate if you tell me how you fix that.
Thank you

**Comment by nitmir** [comment]
Maybe a timezone issue
Beeing on UTC+1 the minimal age is 1 hour

**Comment by TPBT-OFFICIAL** [comment]
@samuel9554 @nitmir Yes i think its a timezone issue but im not completly sure how to fix it.

**Comment by samuel9554** [comment]
@TPBT-OFFICIAL  @nitmir I just changed my timezone in my server to UTC and it doesn't change nothing on the page, same issue. Any idea where to go look to fix that?

**Comment by samuel9554** [comment]
@TPBT-OFFICIAL @nitmir Sorry for the last comment it did work I put it to UTC and now it show 43 min. Thank you guys for the advice. http://screencast.com/t/kqoQ81mVu

**Comment by TPBT-OFFICIAL** [comment]
@samuel9554 Your welcome! :)

**Comment by TPBT-OFFICIAL** [comment]
New Server for Public Database!

**Comment by TPBT-OFFICIAL** [comment]
Database info updated! Sphinx will still be on the same server so no need to update that but new database hosting now!

**Comment by TPBT-OFFICIAL** [comment]
Updated IP'S for database and Sphinx.

**Comment by yatznet** [comment]
@TPBT-OFFICIAL you might want to rethink what you have done your own site is coming up with 

"CDbConnection failed to open the DB connection: SQLSTATE[HY000] [2013] Lost connection to MySQL server at 'reading initial communication packet', system error: 111" when search is used on your hompage

and my site thats connected with your new settings is displaying 404 shit happens whenever a user uses search.on the homepage both your site and mine were working fine until you moved sphinx

please take a look :) thank you

**Comment by TPBT-OFFICIAL** [comment]
@yatznet check again had some problems with iptables :)

**Comment by samuel9554** [comment]
@TPBT-OFFICIAL 
I'm still experiencing the same issues that @yatznet have mentioned

**Comment by yatznet** [comment]
@TPBT-OFFICIAL i can conform this is indeed working again. thank you :)

**Comment by yatznet** [comment]
@TPBT-OFFICIAL i think you need to re-enable your indexing script for sphinx. the one thats ment to run every 5hours because its not run since you fixed your iptables 17hours ago

**Comment by TPBT-OFFICIAL** [comment]
@yatznet your right! i changed server so i forgot to reenable a few things.

**Comment by TPBT-OFFICIAL** [comment]
Database and Sphinx Information have been updated! My domain was suspended and is pending legal action (don't really know what that means) so i got a new domain tpbt.ch

**Comment by yatznet** [comment]
not been able to access your site for over a week here anyway my isp blocked it ironically though it still allowed my site despite it using the same database. and dude you cant close your site at the end of april or alot of us here will cry :'(

**Comment by TPBT-OFFICIAL** [comment]
@yatznet Are you able to access https://tpbt.ch ? Also whats your ISP? just a personal question.
Well the thing is i can't get enough funds :( im starting to think the only solution here would be start charging people for access to the database just not as expensive as you said maybe just 6$ per month 18$ per year, 60$ permenant just in case people wanted to choose. Or just keep on going and hope that ill get enough traffic on the website that ill get enough funds but if i don't ill have to shut it down.

**Comment by TheChiefCoC** [comment]
If you want I can host the databas and site for you for free. :) As I use the original pirate bay now , I no longer use this.

**Comment by TPBT-OFFICIAL** [comment]
@TheChiefCoC The website and the database are actually not the most expensive part, the most expensive part is the SPHINX database because it uses a lot of RAM.

**Comment by TheChiefCoC** [comment]
I have a 8 GB RAM is it enough?

**Comment by TPBT-OFFICIAL** [comment]
That is surely enough! :) Would you be able to host SPHINX for me? also whats the location?

**Comment by samuel9554** [comment]
I have 2 server running I can host a redundant sphinx in case one go down. I'm from Canada

**Comment by yatznet** [comment]
@TPBT-OFFICIAL My isp is Virgin Media and yes i can access your new url. atleast until they realize you moved and block that as-well. but that's why we have the tor network for :)

well i could  access it earlier but now i get this
"Anti-DDoS Filter cannot connect to protected host.
Most likely the firewall (CSF?) on protected host blocks filter IP addresses due to a large number of requests.
If you are the administrator, please disable the firewall or whitelist the filter IP's."

are you still using cloudflare? because if you are and alot of traffic is going to your site at once it will assume all the traffic is from the same place as they will all come from cloudflares ip's. basically cloudflare and csf dont work well together on sites with alot of traffic. 

and i'll take the one years access for $18 please :) bargain

**Comment by TPBT-OFFICIAL** [comment]
@yatznet The website was under a DDos attack and the layer7 protection was malfuctioning it seems to be fixed now and i don't use cloudflare anymore since its from the USA and it makes it feel like the website is "being monitored" so i disabled cloudflare and changed from my previous hosting in sweden (If i got DDos Attacked the IP of the server would be nullrouted for 3 hours) to a netherlands DDOS protected host for 2 days and then they advised me to change to their ukrainian datacenter because of it being out of EU and and being a safe place for my website. About the Database i think there should still be a free plan but with limitations like maybe it has a limit of connections at the same time in the database or limit of connections per hour.

**Comment by yatznet** [comment]
oh ok. about the database, i agree but setting it up with limitations for some and no limitations for others isn't going to be a easy task. if you dont mind me asking who is your host? if you dont want to say in a public forum i can drop you a email. i believe i still have your email address from a previous conversation.

**Comment by TPBT-OFFICIAL** [comment]
@yatznet You would have my email address if my domain tpbt.org hadn't been suspended, you can contact me at admin@tpbt.ch but i don't mind saying it in public, the sphinx hosting used to be by a friend of mine on his server but now its on kimsufi and the database is hosted in RackSpace with multiple IP's from different locations to make it harder to block it.
I have actually made a story of TPBT as my current hosting provider asked me a few questions:
February 2013: Domain TPBT.ORG registered by me
September 2013: Used as proxy for thepiratebay till February 2014
September 2013: February 2014: Gained Alexa rank of 89,485th
February 2014: Domain expired
October 2014: Domain registered again by me and used as a proxy for thepiratebay till 8 December 2014 (TPB Raided)
November 2014: Hit with 400,000+ Url takedowns on google and mentioned in torrentfreak (http://torrentfreak.com/google-takedown-requests-surge-after-new-anti-piracy-measures-140102/) heres the Takedown requests http://www.google.com/transparencyreport/removals/copyright/domains/tpbt.org/
8 December 2014: The Pirate Bay raided and 6-8 days later Openbay Script released.
30 December 2014: Isohunt shutsdown public Openbay Database TPBT.ORG goes down temporarly
31 December 2014: Bought Kimsufi Dedicated Server for Database and Sphinx Hosting and a VPS at yourserver.se (Sweden host) for the frontend website
1 January 2015: I made my own database and Sphinx with torrents from other torrent websites (including an old backup of The Pirate Bay torrent hashes) and made the database public on github for people that didn't have that much money (https://github.com/isohuntto/openbay/issues/142)
8 January 2015: Enabled ADS on the website to help fund the website and database
10 February 2015: Website started getting a lot of money and added torrent scraper to ad torrents automaticly to the database
14 February 2015: Sphinx and Database Separated, Sphinx on Kimsufi Server and Database on rackspace
March 2015: Website started getting less visitors and less funds
30 March 2015: Changed frontend Host from yourserver.se (Sweden) to BlazingFast(Netherlands)

31 March 2015: LOTS OF THINGS (bad things) happened!
- tpbt.org domain suspended (Don't know why yet)
- tpbt.org ssl revoked (Don't know why yet also)
- Registrar domain account deactivated
- sslcertificate account deactivated

1 April 2015: Also a lots of things (good things) happened!
- Bought domain tpbt.ch
- Bought ssl certificate from ssls.com
- Set up https://files.tpbt.ch/ for hosting tpbt images and files and other stuff
- Set up https://the.tpbt.ch (Hosted on online.net ARM Physical Servers) Direct proxy for thepiratebay.

2-3 April 2015: Migrated Server from Netherlands Datacenter at BlazingFast to Ukrainian Datacenter.

**Comment by yatznet** [comment]
wow thats a story and a half. and im being ripped off with my hosting lol all because i insist on having cpanel with my vps

**Comment by TPBT-OFFICIAL** [comment]
@yatznet My VPS has Plesk in it, i used to love cpanel but i realised Plesk is a lot easier to use (at least for me)

**Comment by yatznet** [comment]
@TPBT-OFFICIAL is that the one at blazingfast?

**Comment by TPBT-OFFICIAL** [comment]
@yatznet Yes and its currently getting attacked again "http://prntscr.com/6patoq"
The attack isn't that big currently only +400Mb/s

**Comment by yatznet** [comment]
@TPBT-OFFICIAL atleast its not knocked the site offline this time

**Comment by TPBT-OFFICIAL** [comment]
@yatznet Anti-ddos system seems to be working properly now.

**Comment by yatznet** [comment]
@TPBT-OFFICIAL thats good :)

**Comment by TPBT-OFFICIAL** [comment]
I gotten a massive Bitcoin donation by someone (Anonymous obviously) wich i won't tell the ammout but its more then enough to keep hosting TPBT (and get a few extra "stuff") for at least 6 more months maybe, to whoever has donated thank you very much :)
Database and Sphinx info Updated again!
Reason for update:
Since its possible that some internet providers might block some domais i might as well create a lot of them in case some get blocked.

**Comment by TheChiefCoC** [comment]
Hey @TPBT-OFFICIAL nice xD :santa: glad you are making good use of that. Also would you like to host the server still? can we talk somewhere else?

**Comment by TPBT-OFFICIAL** [comment]
We will get in contact with you soon.

**Comment by TPBT-OFFICIAL** [comment]
Sorry email auto replies :P you can contact me on admin@tpbt.ch

**Comment by TPBT-OFFICIAL** [comment]
New Update for you all and i think its a very good idea to do this :)
Free hosting for openbay/Thepiratebay proxy websites!
I will be setting up a website for the freehosting some time soon don't have that much time right now BUT if you wish to have your website hosted on my offshore server with ANTI-DDOS protection Unlimited bandwidht (limited disk space but we can talk about it in our email) with Plesk control panel please send me an email at admin@tpbt.ch ill try to set up an account for you ASAP :)

**Comment by TPBT-OFFICIAL** [comment]
And another domain has been suspended TPBT.RE
